### DISCUSSION TITLE
Specify id/relay server as part of target id when connecting to another device

### DISCUSSION THREAD

**Discussion by vladimir-angelov-1337** [rustdesk/rustdesk#2072] [Ideas]
It would be cool if instead of "123 456 789" I could write "123 456 789@domain.com" when connecting to another device and that would use domain.com as the id/relay server for this specific connection. That way even with self-hosted setups we could connect cross-organization.

**Comment by AtanasDDimitrov** [comment]
Yes! This would be very useful!

**Comment by likedigital-slavik** [comment]
That sounds way more efficient and safety. Definitely will use it!

**Comment by ivanov-GSGroup** [comment]
Yes! This can be useful for us!

**Comment by vladimir-angelov-1337** [comment]
Come to think of it, would be best if we could somehow use emails instead of numbers. We're willing to help by contributing with an MR but need some help going through the source code.

**Comment by IliyanAng** [comment]
Up.

This will add great flexibility.

**Comment by animus95** [comment]
This would be an awesome feature.

**Comment by simeonovPavel** [comment]
Please implement this it would be great !!!

**Comment by PetyaPamukliyska** [comment]
It will be super helpful, think about it!

**Comment by borislavvelkov** [comment]
It will be really cool, yes!

**Comment by anojoyman** [comment]
This would be good addition. Also to expand on the idea there may be a special group when you use private id/relay server to use this server for that group and public/default for others. Or define id/relay server for each group to use, which I think would be great. This way you can use multiple id/relay servers with ease.

**Comment by HFord1234** [comment]
Please make it happen, so we can finally ditch anydesk

**Comment by marto7777777** [comment]
It would be very beneficial!

**Comment by Stanwolf157** [comment]
Definitely needed as soon as possible!

**Comment by rustdesk** [comment]
Make sense #2118

**Comment by pedjas** [comment]
This is necessity. As if you have several clients set to connect to several hosts and they all share the same configuration it becomes quite a mess. You have all remotes listed at the same place regardless they are not of the same server. There is no clue to which server one has to connect to be able to connect specific remote client.

**Comment by TorstenVo** [comment]
It would be very useful and ideally, the planned address book would be connected to the respective server/domain.
Appreciate your great work.

**Reply by rustdesk** [reply]
Then how do you specify key?

**Reply by pedjas** [reply]
Since key is already public (it is written in file name, openly visible, and all clients use the same key) then key also may be provided in target id.

Even better, it would be great that keys are not part of connection parameters. It would be much better if client connects to server, gets key from it and then switch to encrypted connection using provided key. HTTPS works that way.

**Comment by d3sync** [comment]
Indeed some kind of federation would be awesome for selfhosted relays / id servers
